tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/IbmMq/IbmMqHeadersAdapter.cs

```diff
@@ -55,28 +55,48 @@ public void Set(string name, string value)
     {
         var normalizedName = NormalizeName(name);
         RemoveNormalized(normalizedName);
-        _message.SetStringProperty(normalizedName, value);
+        var val = StringToUnsignedBytes(value);
+        message.SetBytesProperty(normalizedName, val);
     }
 
     public void Add(string name, string value)
     {
-        var normalizedName = NormalizeName(name);
-        RemoveNormalized(normalizedName);
-        _message.SetStringProperty(normalizedName, value);
+        Set(name, value);
     }
 
     private void RemoveNormalized(string normalizedName)
     {
         try
         {
-            _message.DeleteProperty(normalizedName);
+            message.DeleteProperty(normalizedName);
         }
         catch
         {
             // ignored
         }
     }
 
+    private string StringFromSignedBytes(sbyte[] buf)
+    {
+        // since the text is ASCII signed and unsigned bytes are the same (0-127)
+        var unsigned = new byte[buf.Length];
+        Buffer.BlockCopy(buf, 0, unsigned, 0, buf.Length);
+
+        return Encoding.ASCII.GetString(unsigned);
```

Comment:
You can use `Unsafe.As<>` from `System.Runtime.CompilerServices` on non .NET FX and we have a vendored version on .NET FX.
```suggestion
        return Encoding.ASCII.GetString(Unsafe.As<byte[]>(unsigned));
```